### PR TITLE
zero extension expression

### PR DESCRIPTION
**Pull request by kroening**
- [X] Each commit message has a non-empty body, explaining why the change was made.
- [X] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by tautschnig — APPROVED** [review]
I would be nice to see support in the simplifier for any case that does zero_extend of a constant.

**Comment by tautschnig** [comment]
> I would be nice to see support in the simplifier for any case that does zero_extend of a constant.

I believe this is actually a necessary feature for unit tests to pass.

**Comment by thomasspriggs** [comment]
I think the support for zero_extend should be added to the simplifier before merging. Moving from expressions which are supported by the simplifier to a new one which is not may have a performance cost.

**Comment by kroening** [comment]
Now with simplifier support.

File: src/util/bitvector_expr.cpp [hunk 1]
```diff
@@ -279,3 +278,19 @@ exprt find_first_set_exprt::lower() const
...
+  const auto new_width = to_bitvector_type(type()).get_width();
+
+  if(new_width > old_width)
+  {
+    return concatenation_exprt{
+      bv_typet{new_width - old_width}.all_zeros_expr(), op(), type()};
+  }
+  else // new_width <= old_width
+  {
+    return extractbits_exprt{op(), integer_typet{}.zero_expr(), type()};
```
Comment (peterschrammel) [inline src/util/bitvector_expr.cpp]: Isn't the index argument a sizet?

**Review by peterschrammel — APPROVED** [review]

**Review by martin-cs — APPROVED** [review]
Looks good.

**Pull request merged**